tests/Elasticsearch/Tests/YamlRunnerTest.php

```diff
@@ -77,727 +67,621 @@ public static function setUpBeforeClass()
-        foreach ($pieces as $piece) {
-            $piece = str_replace('\.', '.', $piece);
-            if (!is_array($context) || !array_key_exists($piece, $context)) {
-                // error occurred
-                $this->log("Could not find nested property [$piece] in context");//.print_r($context, true);
-                $this->log("\nReturning null...");
-                return null;
+        if (empty($context)) {
+            return $data;
+        }
+
+        if (is_string($data) && preg_match('/\$.+?$/', $data)) {
+            if (!array_key_exists($data, $context)) {
+                static::fail(sprintf('Cannot find variable %s in context', $data));
             }
-            $context = &$context[$piece];
+
+            return $context[$data];
+        }
+
+        if (!is_array($data) && !$data instanceof \stdClass) {
+            return $data;
+        }
+
+        foreach ($data as $key => &$value) {
+            $value = $this->replaceWithContext($value, $context);
         }
 
-        return $context;
+        return $data;
     }
 
     /**
-     * Really ugly hack until upstream Yaml date parsing is fixed
-     * See: https://github.com/symfony/symfony/issues/8580
-     * TODO
+     * Resolve a value into an array given a specific field
      *
-     * @param $file
-     * @param $document
+     * @param $result
+     * @param $field
      *
      * @return mixed
      */
-    private function checkForTimestamp($file, $document)
+    private function resolveValue($result, $field, &$context)
     {
-        $isMatch = preg_match($this->getTimestampRegex(), $document, $matches);
-        if ($isMatch) {
-            $newTime = new \DateTime($matches[0].'GMT');
-            $document = preg_replace($this->getTimestampRegex(), $newTime->format('U') * 1000, $document);
+        if (empty($field)) {
+            return $result;
         }
 
-        return $document;
+        foreach ($context as $key => $value) {
+            $field = preg_replace('/('.preg_quote($key, '/').')/', $value, $field);
+        }
+
+        $operationSplit = explode('.', $field);
+        $value = $result;
+
+        do {
+            $key = array_shift($operationSplit);
+
+            if (substr($key, -1) === '\\') {
+                $key = substr($key, 0, -1) . '.' . array_shift($operationSplit);
+            }
+
+            if (!is_array($value)) {
+                return $value;
+            }
+
+            if (!array_key_exists($key, $value)) {
+                return false;
+            }
+
+            $value = $value[$key];
+        } while (count($operationSplit) > 0);
+
+        return $value;
     }
 
     /**
-     * Hack to rewrite the command `task.list` into `task.get`, since list is a reserved
-     * word in PHP. :/
+     * Format a regex for PHP
      *
-     * @param $file
-     * @return mixed
+     * @param $regex
+     *
+     * @return string
      */
-    private function checkForList($document)
```

Comment:
This `checkForList()` method will need to be replicated in the new runner, somehow.  The `tasks.list/10_basic.yaml` invokes the [Tasks.List API](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/tasks.list.json), which isn't compatible with PHP (since the method call would be `$client->tasks()->list([...])`, and `list()` is a reserved PHP function).  

The method is renamed `get()` in the client, and we have to do the substitution before calling.